common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -155,11 +156,22 @@ public Random current() {
             }
         }
         DIRECT_MEMORY_LIMIT = maxDirectMemory;
-        logger.debug("io.netty.maxDirectMemory: {} bytes", maxDirectMemory);
+        logger.debug("-Dio.netty.maxDirectMemory: {} bytes", maxDirectMemory);
+
+        int tryAllocateUninitializedArray =
+                SystemPropertyUtil.getInt("io.netty.tryAllocateUninitializedArray", 1024);
+        TRY_ALLOCATE_UNINITIALIZED_ARRAY = javaVersion() >= 9 && PlatformDependent0.hasAllocateArrayMethod() ?
+                tryAllocateUninitializedArray : -1;
+        logger.debug("-Dio.netty.tryAllocateUninitializedArray: {}", tryAllocateUninitializedArray);
```

Comment:
@normanmaurer if you use `Integer.MAX_VALUE` instead of `-1` as the opt-out case (i.e. never have to worry about negative values), then the condition below will become simpler:

```java
return size > TRY_ALLOCATE_UNINITIALIZED_ARRAY ?
    PlatformDependent0.allocateUninitializedArray(size) :
    new byte[size];
```